Rust Foundation: July 2024 Alpha-Omega Update (#9)
JoelMarcey/alpha-omega: alpha/engagements/2024/Rust Foundation/update-2024-07.md

> There has been continued progress on verifying the repos that crates say they are associated with. We are also nearly complete on moving our security to the cloud in order to have more consistent runs, analysis and data.

Author: @walterhpearce @LawnGnome If there is anything specific to allude to in the cloud infrastructure migration, please let me know. I will add a section for it in the Engineering section.

## Engineering

### Crate repo verification progress

> Adam has continued working on[provenance](https://lawngno.me/blog/2024/06/10/divine-provenance.html) tracking, verifying that a given crate is actually associated with the repository it claims to be. He has now analyzed the top 5000 crates by download count and has found no obvious red flags.Verification now infers commits based on tag data if the crate file was packaged without VCS info. This finds more mismatches as a result, but they’re all innocent — build artifacts, swap files, etc. Up next is to analyze the entire crates.io corpus.

Reviewer: ```suggestion
Adam has continued working on [provenance](https://lawngno.me/blog/2024/06/10/divine-provenance.html) tracking, verifying that a given crate is actually associated with the repository it claims to be. He has now analyzed the top 5000 crates by download count and has found no obvious red flags. Verification now infers commits based on tag data if the crate file was packaged without VCS info. This finds more mismatches as a result, but they’re all innocent — build artifacts, swap files, etc. Up next is to analyze the entire crates.io corpus.
```

### Crate deletion RFC

> The crate deletion [RFC](https://github.com/rust-lang/rfcs/pull/3660) is now in FCP. Recall, Tobias wrote this RFC to provide crate owners with a mechanism to delete crates from crates.io under certain [conditions](https://github.com/Turbo87/rust-rfcs/blob/crates-io/crate-deletions/text/3660-crates-io-crate-deletions.md#proposal).

Reviewer 2: the RFC is not in FCP yet. I've proposed FCP, but the other team members still need to take a look and agree...

Reviewer 2: ```suggestion
The crate deletion [RFC](https://github.com/rust-lang/rfcs/pull/3660) is close to moving into FCP now. Recall, Tobias wrote this RFC to provide crate owners with a mechanism to delete crates from crates.io under certain [conditions](https://github.com/Turbo87/rust-rfcs/blob/crates-io/crate-deletions/text/3660-crates-io-crate-deletions.md#proposal).
```

### crates.io lib/bin detection

> Tobias backfilled `lib/bin` information for existing crate releases. The `has_lib` and `bin_names` columns in the crates.io database have been backfilled see [conversation](https://rust-lang.zulipchat.com/#narrow/stream/318791-t-crates-io/topic/lib.2Fbin.20detection)). For example, https://crates.io/crates/ripgrep finally shows `cargo install` now.

Reviewer: ```suggestion
Tobias backfilled `lib/bin` information for existing crate releases. The `has_lib` and `bin_names` columns in the crates.io database have been backfilled (see [conversation](https://rust-lang.zulipchat.com/#narrow/stream/318791-t-crates-io/topic/lib.2Fbin.20detection)). For example, https://crates.io/crates/ripgrep finally shows `cargo install` now.
```

### Crate specific RSS feed

> Thanks to Tobias, there is now a [crate-specific RSS feed](https://github.com/rust-lang/crates.io/pull/9064). You can subscribe to get all the details on the specific crates you are interested in. RSS feeds per crate are published at https://static.crates.io/rss/crates/{name}.xml. The feeds are synced with the database in a background job after every successful publish of the corresponding crate. It includes the latest 10 releases of the crate, but at least all releases within the last 24 hours.

Reviewer 2: ```suggestion
Thanks to Tobias, there are now [crate-specific RSS feeds](https://github.com/rust-lang/crates.io/pull/9064). You can subscribe to get all the details on the specific crates you are interested in. RSS feeds per crate are published at https://static.crates.io/rss/crates/{name}.xml. The feeds are synced with the database in a background job after every successful publish of the corresponding crate. It includes the latest 10 releases of the crate, but at least all releases within the last 24 hours.
```

> Thanks to Tobias, there is now a [crate-specific RSS feed](https://github.com/rust-lang/crates.io/pull/9064). You can subscribe to get all the details on the specific crates you are interested in. RSS feeds per crate are published at https://static.crates.io/rss/crates/{name}.xml. The feeds are synced with the database in a background job after every successful publish of the corresponding crate. It includes the latest 10 releases of the crate, but at least all releases within the last 24 hours.

Reviewer: ```suggestion
Thanks to Tobias, there is now a [crate-specific RSS feed](https://github.com/rust-lang/crates.io/pull/9064). You can subscribe to get all the details on the specific crates you are interested in. RSS feeds per crate are published at https://static.crates.io/rss/crates/{name}.xml (example: https://static.crates.io/rss/crates/serde.xml). The feeds are synced with the database in a background job after every successful publish of the corresponding crate. It includes the latest 10 releases of the crate, but at least all releases within the last 24 hours.
```

We are pleased to [announce](https://foundation.rust-lang.org/news/welcoming-infrastructure-engineer-marco-ieni-to-the-rust-foundation-team/) that Marco Ieni has joined the Rust Foundation as the newest member of our engineering team! Marco joins the Rust Foundation as its second Infrastructure Engineer, providing much-needed support to the Rust project and its infrastructure. As a member of the engineering team, Marco will ensure that the infrastructure supporting Rust and related projects is reliable, efficient, productive, and secure. He will be responsible for the operations of all current and future infrastructure, managing existing tools and new developments while working directly with the Rust Project Infrastructure team in their daily activities.

### Sustainability Efforts

Adam continues to focus on project sustainability, focusing on repo and maintainer information. Adam gave a Python specific version of his [Quantifying Nebraska talk](https://www.youtube.com/watch?v=QMHpy_mcx0Q) at [North Bay Python](https://pretalx.northbaypython.org/nbpy-2024/talk/9EXJ7T/). He is developing tooling around this.

### Development of crates.io update

> Tobias [published](INSERT LINK HERE) an update of the development progress of crates.io.

Author: @Turbo87 Just waiting for the link on this one.

Reviewer 2: yeah, unfortunately I still need a review approval from a team lead to be able to merge the blog post

Reviewer: maybe we can link to the PR?
